lib/MusicBrainz/Server/Edit/Role/AllowAmending.pm

```diff
@@ -0,0 +1,47 @@
+package MusicBrainz::Server::Edit::Role::AllowAmending;
+use MooseX::Role::Parameterized;
+
+parameter create_edit_type => ( isa => 'Int', required => 1);
+parameter entity_type => ( isa => 'Str', required => 1);
+
+role {
+    my $params = shift;
+
+    my $create_edit_type = $params->create_edit_type;
+    my $entity_type = $params->entity_type;
+
+    requires 'c', 'editor_id';
+
+=method can_amend
+
+Check if an editor is still allowed to amend an entity, that is,
+if it has been created by the same editor less than an hour ago.
+
+=cut
+
+    method 'can_amend' => sub {
+        my ($self, $amended_entity_id) = @_;
+
+        my $add_entity_edit = $self->c->sql->select_single_value("
+            SELECT id FROM edit
+              JOIN edit_$entity_type ON edit.id = edit_$entity_type.edit
+             WHERE edit_$entity_type.$entity_type = $amended_entity_id
```

Comment:
I'd prefer using `?` placeholders for the values like `AllowAmendingRelease` did (though I think it's probably fine for immutable constants defined in the code, like `$create_edit_type`).